pkg/runctl/run_manager_test.go

```diff
@@ -1170,18 +1170,21 @@ func Test_RunManager_addTektonTaskRunParamsForJenkinsfileRunnerImage(t *testing.
 			defer mockCtrl.Finish()
 			mockPipelineRun := mocks.NewMockPipelineRun(mockCtrl)
 			mockPipelineRun.EXPECT().GetSpec().Return(tc.spec).AnyTimes()
-
+			existingParams := []tekton.Param{
+				tektonStringParam("AlreadyExistingParam1", "foo"),
+			}
 			tektonTaskRun := tekton.TaskRun{
 				Spec: tekton.TaskRunSpec{
-					Params: []tekton.Param{},
+					Params: existingParams,
```

Comment:
You need to pass a deep copy of `existingParams` here. Otherwise you cannot detect modifications to the underlying array elements.

Reply:
Fixed by b151a74.